### DISCUSSION TITLE
How to Get ValidationError back from Validate?

### DISCUSSION THREAD

**Discussion by johnwrobinson** [santhosh-tekuri/jsonschema#131] [Q&A] [answered]
I think I'm being really silly, but I cannot for the life of me work out how to get a ValidationError returned from the Validate function.

Every time it returns a normal Error, and the signature on the function says it only returns Error.

I've tried:

```
...
var vError *jsonschema.ValidationError
...
vError = schema.Validate(f)
```

and my IDE just complains that Validate doesn't return a ValidationError type.

How do I get this to work properly?

Thanks in advance!

**Comment by santhosh-tekuri** [comment, marked as answer]
in golang, function signature always will have `error`. you have to check for specific error type as shown below:

```go
var err = schema.Validate(f)
if verr, ok = err.(*jsonschema.ValidationError); ok {
    fmt.Println(verr.InstanceLocation)
}
```

**Reply by johnwrobinson** [reply]
Thanks for your assistance!

**Comment by johnwrobinson** [comment]
I was able to resolve this issue with:

```
if err := schema.Validate(v); err != nil {
	ve, ok := err.(*jsonschema.ValidationError)
	if ok {
		b, _ := json.MarshalIndent(ve.BasicOutput(), "", "  ")
		fmt.Println(string(b))
	} else {
		handleErr(err)
	}
}
```
